images/capi/packer/azure/scripts/init-sig.sh

```
@@ -98,3 +98,15 @@ az sig image-definition create \
    --sku centos-7.7-gen2 \
    --hyper-v-generation V2 \
    --os-type Linux
+az sig image-definition create \
+   --resource-group ${RESOURCE_GROUP_NAME} \
+   --gallery-name ${GALLERY_NAME} \
+   --gallery-image-definition capi-flatcar-${FLATCAR_CHANNEL}-${FLATCAR_VERSION}-gen2 \
+   --publisher capz \
+   --offer capz-demo \
+   --sku flatcar-${FLATCAR_CHANNEL}-${FLATCAR_VERSION}-gen2 \
+   --hyper-v-generation V2 \
+   --os-type Linux
+
+az vm image terms accept --urn kinvolk:flatcar-container-linux-free:${FLATCAR_CHANNEL}:latest
```

not sure it's a good idea to run this in this script which is common to all SIG definitions and accept license terms on behalf of the user. Perhaps we could split it into a specific flatcar setup script so that it only gets run if the user is trying to create a flatcar image, that way it's at least only run when the user has expressed intent to use the flatcar source images and we don't end up accepting terms on every subscription that runs this, regardless of their need for flatcar. The other option is to not run this in a script and add it as a one-time setup instruction in docs but I can see why that is less attractive.

---

Maybe there is a way to check if terms has been accepted and bail when flatcar target is built? This way we fail early and tell user what to do to get through.

---

the terms for the gen2 image have now been accepted for the sponsorship subscription used by prow jobs so tests should pass without this

---

Dropped this commit.

---

I don't see a way to check if terms have been accepted and the error message from the VM create is clear on what needs to be done and provides links to Azure docs.

FWIW: we would all prefer that the user not have to agree to any terms to use any free Flatcar images, but this is not something that the Azure marketplace allows us to express.